Dockerfiles/agent/amd64/Dockerfile

```diff
@@ -169,7 +171,7 @@ RUN tar xzf s6.tgz -C / --exclude="./bin" \
  && chmod g+r,g+w,g+X -R /etc/datadog-agent/ /etc/s6/ /var/run/s6/ /var/log/datadog/ \
  && chmod 755 /probe.sh /initlog.sh \
  && chown root:root /readsecret.py /readsecret.sh /readsecret_multiple_providers.sh \
- && chmod 500 /readsecret.py /readsecret.sh /readsecret_multiple_providers.sh
+ && chmod 550 /readsecret.py /readsecret.sh /readsecret_multiple_providers.sh
```

Comment:
Hi @urnfdog !
For a bit more context, this same image is used for both the `datadog-agent` `datadog-agent-cluster-worker` . In theory they have the same capabilities, although in practice we use the `cluster-worker` just to run cluster-level checks that don't require `root` privileges. That's why it's run as `dd-agent`. However, the containerized `datadog-agent` is run as `root` for a few reasons (access to container log files and use of kernel probes, for example).

To answer your questions:
> Who else is in the root group other than dd-agent which the cluster check runners are running as

There are no other users in this group.

> What service runs this as root? Should it be run as dd-agent instead? Should the owner for the file itself be set to dd-agent instead of changing permission bits? I notice that on line 170 we chown to dd-agent:root.

Due to constraints we faced in other environments (e.g. OpenShift) we actually do plan to change this in the near future - to set the default user in the image to `dd-agent` and use a SecurityContext to use the `root` user in the situations it's needed.

> Digging back into git history around this we find this PR from 4 years ago... Are these still the current security recommendations for the agent?

Yes I believe so.

Reply:
Also I should mention that the `cluster-worker` currently runs as `root`, and this change is an effort to support running it as `dd-agent` instead (and not break use of the secret backend feature).